hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -1207,6 +1213,16 @@ func (r *HostedClusterReconciler) reconcile(ctx context.Context, req ctrl.Reques
 		}
 	}
 
+	// Ensure the CAPI deployments have finalizers
+	if hcp != nil {
+		_, machineManagementDisabled := hcp.Annotations[hyperv1.DisableMachineManagement]
+		if !machineManagementDisabled {
```

Comment:
we could pass the `hcluster` object to the function and let it return early if the annotation is set, instead of checking it every time before calling it.

Reply:
All annotation are set on the HostedCluster and then copied to the HCP, nothing is set directly on the hcp by users.

> Also if we do it like this I am assuming we do not need to nil-check the hcp.

yep